packages/inference/src/providers/replicate.ts

```diff
@@ -14,37 +14,142 @@
  *
  * Thanks!
  */
-import type { BodyParams, HeaderParams, ProviderConfig, UrlParams } from "../types";
-
-export const REPLICATE_API_BASE_URL = "https://api.replicate.com";
-
-const makeBaseUrl = (): string => {
-	return REPLICATE_API_BASE_URL;
-};
-
-const makeBody = (params: BodyParams): Record<string, unknown> => {
-	return {
-		input: params.args,
-		version: params.model.includes(":") ? params.model.split(":")[1] : undefined,
-	};
-};
-
-const makeHeaders = (params: HeaderParams): Record<string, string> => {
-	return { Authorization: `Bearer ${params.accessToken}`, Prefer: "wait" };
-};
-
-const makeUrl = (params: UrlParams): string => {
-	if (params.model.includes(":")) {
-		/// Versioned model
-		return `${params.baseUrl}/v1/predictions`;
-	}
-	/// Evergreen / Canonical model
-	return `${params.baseUrl}/v1/models/${params.model}/predictions`;
-};
-
-export const REPLICATE_CONFIG: ProviderConfig = {
-	makeBaseUrl,
-	makeBody,
-	makeHeaders,
-	makeUrl,
-};
+import { InferenceOutputError } from "../lib/InferenceOutputError";
+import { isUrl } from "../lib/isUrl";
+import type { BodyParams, HeaderParams, InferenceTask, UrlParams } from "../types";
+import { omit } from "../utils/omit";
+import { TaskProviderHelper } from "./providerHelper";
+
+export interface ReplicateOutput {
+	output?: string | string[];
+}
+
+export class ReplicateTask extends TaskProviderHelper {
+	constructor(task: InferenceTask, url?: string) {
+		super("replicate", url || "https://api.replicate.com", task);
+	}
+
+	makeRoute(params: UrlParams): string {
+		if (params.model.includes(":")) {
+			return "v1/predictions";
+		}
+		return `v1/models/${params.model}/predictions`;
+	}
+	preparePayload(params: BodyParams): Record<string, unknown> {
+		return {
+			input: {
+				...omit(params.args, ["inputs", "parameters"]),
+				...(params.args.parameters as Record<string, unknown>),
+				prompt: params.args.inputs,
+			},
+			version: params.model.includes(":") ? params.model.split(":")[1] : undefined,
+		};
+	}
+	override prepareHeaders(params: HeaderParams, binary: boolean): Record<string, string> {
+		const headers: Record<string, string> = { Authorization: `Bearer ${params.accessToken}`, Prefer: "wait" };
+		if (!binary) {
+			headers["Content-Type"] = "application/json";
+		}
+		return headers;
+	}
+
+	override makeUrl(params: UrlParams): string {
+		if (params.model.includes(":")) {
+			return `${params.baseUrl}/v1/predictions`;
+		}
+		return `${params.baseUrl}/v1/models/${params.model}/predictions`;
+	}
+
+	override getResponse(response: unknown): unknown {
+		void response;
+		throw new Error("Method not implemented.");
+	}
```

Comment:
dunno how JS works but I'd except we don't have to override this one if we make the base class abstract